### ISSUE TITLE
Incorrect Detection of Roadside Trees as Obstacles on the Road (Autoware.universe+Carla 0.9.14 co-simulation)

### ISSUE THREAD

**Issue by sgNicola** [autowarefoundation/autoware.universe#6940]
### Checklist

- [X] I've read the [contribution guidelines](https://github.com/autowarefoundation/autoware/blob/main/CONTRIBUTING.md).
- [X] I've searched other issues and no duplicate issues were found.
- [X] I'm convinced that this is not my fault but a bug.

### Description

Where trees located beside roads are incorrectly considered as obstacles on the road. It disrupts normal vehicle operations by causing unnecessary avoidance maneuvers or stops.

### Expected behavior

The vehicle should go ahead.

### Actual behavior

The vehicle stops and does not proceed upon approaching a tree, mistaking it for an obstacle on the road.

### Steps to reproduce

1. Launch Autoware.Universe.
2. Drive the vehicle towards areas around the tree
3. Observe that the vehicle stops,  treating it as an obstacle on the road.

### Versions

-OS: Ubuntu 20.04
-ROS2: ros2 galactic
-Autoware: Autoware.universe galactic

### Possible causes

This behavior likely originates from the Surround Obstacle Checker module, which subscribes to the /perception/obstacle_segmentation/pointcloud topic. Furthermore, the absence of detected objects in the /perception/object_recognition/objects topic output may exacerbate the issue, indicating potential malfunctions in the perception module. 

### Additional context

This may have the same root cause as #6936.

**Comment by sgNicola** [comment]
Hi, I would like to upload the ros2bag for steps to reproduce, however, it doesn't support db3 file type.
![Uploading tree_rviz_view.png…]()
![tree_vehicle_view](https://github.com/autowarefoundation/autoware.universe/assets/115910640/e5b76dc1-ac55-4606-a20f-75c24e89092a)
![2024-05-07_21-34](https://github.com/autowarefoundation/autoware.universe/assets/115910640/45a290cd-5d7d-4c8f-bb91-f619f4380e13)

**Comment by badai-nguyen** [comment]
@sgNicola FYI: I am working on kind of similar issue here https://github.com/autowarefoundation/autoware.universe/issues/5594

**Comment by amadeuszsz** [comment]
@badai-nguyen 

Since #5594 is solved, can we close this issue? Does new feature handle these cases?

**Comment by amadeuszsz** [comment]
Closing due to inactivity and issue might be already solved. Feel free to reopen this issue if needed.

**Comment by badai-nguyen** [comment]
> @badai-nguyen
> 
> Since #5594 is solved, can we close this issue? Does new feature handle these cases?

@amadeuszsz I appologize for missed your mention. 
I closed the issue https://github.com/autowarefoundation/autoware.universe/issues/5594  since all necessary changes for the task were merged.
However, I'm not sure the new filter can handle this iusse in simulation environment since new filter mostly denpends on the performance of yolox model for image semantic segmentation.
